evalml/pipelines/components/transformers/encoders/ordinal_encoder.py

```diff
@@ -0,0 +1,289 @@
+                if not isinstance(ltype, Ordinal):
+                    raise TypeError(
+                        f"Column {col} specified in features_to_encode is not Ordinal in nature",
+                    )
+
+            # Put features_to_encode in the same relative order as the columns in the dataframe
+            self.features_to_encode = [
+                col for col in X.columns if col in self.features_to_encode
+            ]
+
+        X_t = X
+
+        ww_logical_types = X.ww.logical_types
+        categories = []
+        if len(self.features_to_encode) == 0:
+            # No ordinal features present - no transformation can take place so return early
+            return self
+        elif self.parameters["categories"] is not None:
+            # Categories specified - make sure they match the ordinal columns
+            input_categories = self.parameters["categories"]
+
+            if len(input_categories) != len(self.features_to_encode):
+                raise ValueError(
+                    "Categories argument must contain as many elements as there are Ordinal features.",
+                )
+
+            if not all(isinstance(cats, list) for cats in input_categories):
+                raise ValueError(
+                    "Each element of the categories argument must be a list.",
+                )
+            # Categories, as they're passed into SKOrdinalEncoder should be in the same order
+            # as the data's Ordinal.order categories even if it's a subset
+            for i, col_categories in enumerate(input_categories):
+                categories_order = ww_logical_types[self.features_to_encode[i]].order
+
+                ordered_categories = [
+                    cat for cat in categories_order if cat in col_categories
+                ]
+                categories.append(ordered_categories)
+        else:
+            # Categories unspecified - use ordered categories from a columns' Ordinal logical type
+            for col in X_t[self.features_to_encode]:
+                ltype = ww_logical_types[col]
+                # Copy the order list, since we might mutate it later by adding nans
+                # and don't want to impact the Woodwork types
+                categories.append(ltype.order.copy())
+
+        # Add any null values into the categories lists so that they aren't treated as unknown values
+        # This is needed because Ordinal.order won't indicate if nulls are present, and SKOrdinalEncoder
+        # requires any null values be present in the categories list if they are to be encoded as
+        # missing values
+        for i, col in enumerate(X_t[self.features_to_encode]):
+            if X_t[col].isna().any():
+                categories[i] += [np.nan]
+
+        encoded_missing_value = self.parameters["encoded_missing_value"]
+        if encoded_missing_value is None:
+            encoded_missing_value = np.nan
+
+        self._encoder = SKOrdinalEncoder(
```

Comment:
This should be saved as the `_component_obj` to match the pattern of the rest of our components!

Reply:
Changing! 

Out of curiosity: Is this out of convention or is it needed to work properly in evalml? I ask, because this was another holdover from the onehot encoder, and I do see `_encoder` used elsewhere across the repo (though not nearly as much as `_component_obj`)

Reply:
It's mostly a convention thing, afaik